app/connectors/CalculationListConnector.scala

```diff
@@ -99,4 +108,28 @@ class CalculationListConnector @Inject()(val http: HttpClient,
       }
     }
   }
+
+  def overwriteCalculationList(nino: Nino, taxYearRange: String, crystallisationStatus: String)
+                              (implicit headerCarrier: HeaderCarrier): Future[Unit] = {
+
+    http.GET[HttpResponse](getOverwriteCalculationListUrl(nino.value, taxYearRange, crystallisationStatus))(
+      httpReads,
+      headerCarrier.withExtraHeaders("Accept" -> "application/vnd.hmrc.2.0+json"),
+      ec
+    ) flatMap { response =>
+      response.status match {
```

Comment:
flatMap usage is fine in this context

Reply:
> Flat mapping is redundant here - you can do a map, return Unit in the first branch, and throw the exception in the second.

like this?
Since it is wrapped in a future would this still be handled in the Controller recover block?
    http.GET[HttpResponse](getOverwriteItsaStatusUrl(nino.value, taxYearRange, itsaStatus))(
      httpReads,
      headerCarrier.withExtraHeaders("Accept" -> "application/vnd.hmrc.2.0+json"),
      ec
    ) map { response =>
      response.status match {
        case OK =>
          (): Unit
        case _ =>
          throw new Exception(s"Overwrite unsuccessful. ~ Response status: ${response.status} ~. < Response body: ${response.body} >")
      }
    }

Reply:
I see, both solutions is fine to me, you can go with the options which generates less code